### PR TITLE
feat(triplet_store): add Altair Anzo triplet store backend

### PR DESCRIPTION
**Pull request by KaifAhmad1**
## Summary

Implements #813 — adds `AnzoStore` as a fourth peer triplet-store backend alongside `BlazegraphStore`, `RDF4JStore`, and `JenaStore`.

- **`semantica/triplet_store/anzo_store.py`** (new): `AnzoStore` class matching the existing backend interface (`execute_sparql`, `add_triplet(s)`, `get_triplets`, `delete_triplet`, `bulk_load`). Anzo speaks plain SPARQL 1.1 over HTTP, so no new dependency is required.
  - Constructor takes `endpoint`, `dataset_uri` (required — Anzo's equivalent of a namespace, but a full URI rather than a short name), `store_type` (default `"graphmart"`, also accepts `"dataset"`), `username`/`password` (HTTP Basic), and `timeout`.
  - Endpoint path is built as `<endpoint>/sparql/<store_type>/<url-encoded_dataset_uri>`, percent-encoding the dataset/graphmart URI — the one real structural difference from Blazegraph's namespace or RDF4J's repository ID.
  - Reuses `sparql_escaping.py` for literal escaping, datatype-IRI resolution, and CONSTRUCT-query detection rather than reimplementing them; `execute_sparql`'s bindings/CONSTRUCT result contract matches `BlazegraphStore` exactly.
  - Bulk loading uses SPARQL `INSERT DATA` (same as `BlazegraphStore`), not Anzo's separate HTTP Client Interface — keeping the `bulk_load()` contract identical across backends.
- **`triplet_store.py`**: added `"anzo"` to `SUPPORTED_BACKENDS` and `NAMED_GRAPH_CAPABLE_BACKENDS`, plus a dispatch branch in `_initialize_store_backend()`.
- **`config.py`**: added `TRIPLET_STORE_ANZO_ENDPOINT` → `anzo_endpoint` env var mapping, following the existing `blazegraph_endpoint`/`jena_endpoint`/`rdf4j_endpoint` pattern.
- **`__init__.py`**: exports `AnzoStore`.
- **`triplet_store_usage.md`** / **`CHANGELOG.md`**: usage example and changelog entry.

## Test plan

- [x] `tests/triplet_store/test_anzo_store.py` — 32 new tests (mocked HTTP, no live Anzo instance needed): construction/validation, dataset-URI percent-encoding (including special characters), CONSTRUCT/bindings query paths, bulk load/CRUD, object formatting.
- [x] Full `tests/triplet_store/` suite: 293/293 passing, no regressions in the other three backends.
- [x] End-to-end smoke test through the public `TripletStore` API (mocked HTTP): `add_triplet` → `INSERT DATA` to the correctly-encoded endpoint with Basic auth, `get_triplets` parses SELECT bindings, `execute_query` with CONSTRUCT routes through `QueryEngine` and returns parsed Turtle triples, `delete_triplet` issues `DELETE DATA`.
- [x] `register_store("id", "anzo", endpoint, dataset_uri=...)` and `TRIPLET_STORE_ANZO_ENDPOINT` env var verified working without further code changes.

## Open questions from the issue (deferred, non-blocking)

- Bulk loading uses SPARQL `INSERT DATA` rather than Anzo's HTTP Client Interface, per the issue's own default proposal — can be revisited if a live Anzo instance surfaces a case where INSERT DATA is insufficient.
- No sandbox Anzo instance was available for this PR; all tests use mocked HTTP responses, consistent with how the other three backends were merged.

### CODE REVIEW

**Review by Sameer6305 — APPROVED** [review]
Overall, this is a well-implemented backend addition. I performed a comprehensive review of the implementation, backend integration, API parity, and test coverage.

### Review Summary

**Implementation Validation**
- Verified backend registration, factory integration, configuration flow, and exports.
- Reviewed SPARQL generation, object formatting, datatype resolution, language-tag handling, URI validation, and CONSTRUCT query detection.
- Checked HTTP request handling and response parsing for both graph and bindings query paths.
- Compared behavior with the existing Blazegraph, RDF4J, and Jena backends to identify any inconsistencies or regressions.

**Findings**
- No correctness issues or regressions were identified.
- Backend integration is complete and consistent with the existing triplet store architecture.
- Existing implementation already handled the reviewed edge cases correctly.

**Test Coverage**
While reviewing the implementation, I identified several backend-agnostic regression tests that existed in the Blazegraph suite but were missing for Anzo. I added the missing regression tests covering:
- literal escaping
- language-tag serialization
- datatype validation and resolution
- URI-like literal handling
- CONSTRUCT query detection edge cases
- Turtle parsing with valid literals containing braces

These additions are **test-only** and required **no production code changes**.

### Validation
- ✅ `tests/triplet_store/test_anzo_store.py`
- ✅ `tests/triplet_store` (311 passing, 0 failures)

The implementation looks solid, and the additional regression coverage further strengthens confidence in future changes.

**Approved.**

File: semantica/triplet_store/anzo_store.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+        obj = triplet.object
+        metadata = triplet.metadata or {}
+
+        if self._is_uri_value(obj):
+            if obj.startswith("<") and obj.endswith(">"):
+                inner = obj[1:-1]
+                if " " in inner or ">" in inner:
+                    raise ValueError(f"IRI contains invalid characters: {obj!r}")
+                return obj
+            return f"<{obj}>"
```
[the remark this answers is not in the document]
Comment (Sameer6305) [reply semantica/triplet_store/anzo_store.py:389]: Reviewed this in detail against the shared object-formatting implementation and related validation logic.

The current behavior is intentional and consistent with the existing backend implementations. I couldn't reproduce a correctness or injection issue from this code path, and the implementation already passes the relevant regression coverage.

Marking this as resolved.

File: semantica/triplet_store/anzo_store.py [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+            if result_format == "construct":
+                self.progress_tracker.update_tracking(
+                    tracking_id, message="Sending CONSTRUCT query to Anzo endpoint..."
+                )
+                response = requests.post(
+                    sparql_endpoint,
+                    data={"query": query},
+                    headers={
+                        "Content-Type": "application/x-www-form-urlencoded",
+                        "Accept": "text/turtle",
```
[the remark this answers is not in the document]
Comment (Sameer6305) [reply semantica/triplet_store/anzo_store.py:208]: Verified this against the implementation and the Turtle media type specification.

`text/turtle` is the standard media type for Turtle responses and is the appropriate value here. The implementation is consistent with the existing backend behavior, so no change is required.

Resolving this comment.
